### PR TITLE
[test][enhance][refactor] support suite block to specify multiple group, suppo…

### PR DESCRIPTION
**Pull request by 924060929**
# Proposed changes

1. support suite block to specify multiple groups.
2. `TestAction` support compare result to iterator, local file and http stream.
3. support print teamcity service message.
4. abandon the logical:  generate groovy file for sql file
5. support 3 levels parrallel: script file, suite block, thread action
6. support specify JAVA_OPTS for boot shell
7. avoid jvm metaspace oom
8.  use `-d` to run the suite in some directories, instead of `-g`. and `-g`  is used to specify  `groups`

already update demo in the chinese document and demo suites

## Checklist(Required)

1. Does it affect the original behavior: (Yes)
2. Has unit tests been added: (No Need)
3. Has document been added or modified: (Yes)
9. Does it need to update dependencies: (Yes)
10. Are there any changes that cannot be rolled back: (Yes)

@EmmyMiao87

### CODE REVIEW

File: docs/zh-CN/developer-guide/regression-testing.md [hunk 1]
```diff
@@ -489,4 +531,11 @@ thread, lazyCheck, events, connect, selectUnionAll
 
 # 使用查询结果自动生成sql_action用例的.out文件，如果.out文件存在则覆盖
 ./run-regression-test.sh --run sql_action -forceGenOut
+```
+
+## CI/CD的支持
+### TeamCity
+可以使用--teamcity开启TeamCity Service Message. `-Dteamcity.enableStdErr=false`可以让错误日志也打印到stdout中，方便按顺序分析日志。
```
Comment (EmmyMiao87) [inline docs/zh-CN/developer-guide/regression-testing.md:538]: 如果开启 --teamcity 的话，输出的文件是在哪里？以及在teamcity 上需要什么特殊的配置嘛？

File: regression-test/framework/src/main/groovy/org/apache/doris/regression/util/TeamcityUtils.groovy [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+        String timestamp = formatNow()
+        return "##teamcity[testStdOut name='${suiteContext.flowName}' out='${escape(msg)}' flowId='${suiteContext.flowId}' timestamp='${timestamp}']"
+    }
+
+    static String formatStdErr(SuiteContext suiteContext, String msg) {
+        String timestamp = formatNow()
+        return "##teamcity[testStdErr name='${suiteContext.flowName}' out='${escape(msg)}' flowId='${suiteContext.flowId}' timestamp='${timestamp}']"
+    }
+
+//    static void testSuiteStarted(ScriptContext scriptContext) {
```
Comment (EmmyMiao87) [inline regression-test/framework/src/main/groovy/org/apache/doris/regression/util/TeamcityUtils.groovy:41]: Remove it directly ~

File: run-regression-test.sh [hunk 3]
```diff
@@ -80,20 +84,24 @@ Default config file: \${DORIS_HOME}/regression-test/conf/regression-conf.groovy
...
     CLEAN=0
     RUN=0
+    TEAMCITY=0
     WRONG_CMD=0
     while true; do
         case "$1" in
-            --clean) CLEAN=1 ; shift ;;
-            --run) RUN=1 ; shift ;;
+            --clean)    CLEAN=1 ; shift ;;
+	     --teamcity) TEAMCITY=1 ; shift ;;
```
Comment (EmmyMiao87) [inline run-regression-test.sh:103]:
```suggestion
	    --teamcity) TEAMCITY=1 ; shift ;;
```

File: run-regression-test.sh [hunk 4]
```diff
@@ -80,20 +84,24 @@ Default config file: \${DORIS_HOME}/regression-test/conf/regression-conf.groovy
...
+              --run)      RUN=1 ; shift ;;
```
Comment (EmmyMiao87) [inline run-regression-test.sh:104]:
```suggestion
             --run)      RUN=1 ; shift ;;
```

**Comment by EmmyMiao87** [comment]
The test failed. Maybe you should rebase the code ~

File: docs/zh-CN/developer-guide/regression-testing.md (hunk 1 above)
Comment (924060929) [reply docs/zh-CN/developer-guide/regression-testing.md:538]: 不需要输出文件。除了配置启动命令，teamcity不需要修改什么。框架会打印teamcity的能识别的事件日志到标准输出流，teamcity在启动进程后会自动识别标准输出流中的日志

File: regression-test/framework/src/main/groovy/org/apache/doris/regression/util/TeamcityUtils.groovy (hunk 2 above)
Comment (924060929) [reply regression-test/framework/src/main/groovy/org/apache/doris/regression/util/TeamcityUtils.groovy]: done

File: run-regression-test.sh (hunk 3 above)
Comment (924060929) [reply run-regression-test.sh]: done

File: run-regression-test.sh (hunk 4 above)
Comment (924060929) [reply run-regression-test.sh]: done

**Comment by 924060929** [comment]
> The test failed. Maybe you should rebase the code ~

done

File: docs/zh-CN/developer-guide/regression-testing.md (hunk 1 above)
Comment (EmmyMiao87) [reply docs/zh-CN/developer-guide/regression-testing.md:538]: 嗯我的意思就是把配置 teamcity 流水线的这个步骤说一下。这样大家都可以按照这个步骤配置 ~

Comment (924060929) [reply docs/zh-CN/developer-guide/regression-testing.md]: 改了

**Review by EmmyMiao87 — DISMISSED** [review]
LGTM

**Review by EmmyMiao87 — APPROVED** [review]
LGTM

**Pull request merged**